messaging/app/src/main/java/com/google/firebase/example/messaging/MainActivity.java

```diff
@@ -102,4 +109,41 @@ public void onComplete(@NonNull Task<String> task) {
                 });
         // [END log_reg_token]
     }
+
+    @RequiresApi(33)
+    // [START ask_post_notifications]
+    private void askNotificationPermission() {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.POST_NOTIFICATIONS) ==
+                PackageManager.PERMISSION_GRANTED) {
+            // FCM SDK (and your app) can post notifications.
+        } else if (shouldShowRequestPermissionRationale(Manifest.permission.POST_NOTIFICATIONS)) {
+            // TODO: display an educational UI explaining to the user the features that will be enabled
+            //       by them granting the POST_NOTIFICATION permission. This UI should provide the user
+            //       "OK" and "No thanks" buttons. If the user selects "OK" directly request the permission.
```

Comment:
Suggest a couple commas to place:

"If the user selects "OK," directly request the permission.
            //       If the user selects "No thanks," allow

Reply:
Done!